### ISSUE TITLE
AudioAugmenter loudness-norm disablement leaks across batch rows in add_room_ir_to_batch / add_mic_ir_to_batch

### ISSUE THREAD

**Issue by udsy19** [NVIDIA-NeMo/NeMo#16180]
**Describe the bug**

`AudioAugmenter.add_room_ir_to_batch` and `AudioAugmenter.add_mic_ir_to_batch` (`nemo/collections/speechlm2/parts/augmentation.py`) reassign the `use_loudness_norm` **function parameter** to `False` inside their per-row `for i in range(batch_size)` loop whenever a single row's loudness reading is invalid (silent audio, or an exception from the loudness meter). Python does not reset a parameter between loop iterations, so the reassignment leaks into every subsequent row: once one row in a batch is silent, LUFS loudness normalization is silently disabled for every row after it too, even though those rows have perfectly normal audio. The pattern is duplicated verbatim at `:174-182` (`add_room_ir_to_batch`) and `:258-266` (`add_mic_ir_to_batch`):

```python
for i in range(batch_size):
    if use_loudness_norm and PYLOUDNORM_AVAILABLE:
        ...
        if speech_loudness == float('-inf') or not np.isfinite(speech_loudness):
            use_loudness_norm = False   # reassigns the shared parameter
        except Exception:
            use_loudness_norm = False   # same here
    ...
    if use_loudness_norm and PYLOUDNORM_AVAILABLE:
        ...  # never runs for any row after the first disabling one
```

**Steps/Code to reproduce bug**

```python
import torch
from nemo.collections.speechlm2.parts.augmentation import AudioAugmenter

sr = 16000
t = torch.arange(sr, dtype=torch.float32) / sr
batch = torch.stack([
    torch.zeros(sr),                          # row 0: silent
    0.5 * torch.sin(2 * torch.pi * 440 * t),  # row 1: normal
    0.2 * torch.sin(2 * torch.pi * 220 * t),  # row 2: normal
])
lens = torch.tensor([sr, sr, sr])
AudioAugmenter(sample_rate=sr).add_room_ir_to_batch(batch, lens, "<roomir_dir>", use_loudness_norm=True)
# Row 0 correctly skips loudness normalization (it is silent). Rows 1 and 2
# SHOULD still be loudness-normalized, but row 0 permanently disabled the
# shared `use_loudness_norm` variable, so rows 1 and 2 silently fall back to
# plain RMS-based gain compensation instead.
```

Observed on tip `fe43df3eb9`, with a monkeypatched deterministic loudness meter: the number of `pyln.normalize.loudness(...)` calls across the 3-row batch above is **0** (expected: 2, once per non-silent row). `add_mic_ir_to_batch` shows the identical behaviour.

**Expected behavior**

A silent/invalid row should disable loudness normalization only for *that* row. Other rows in the same batch should still get LUFS loudness normalization when `use_loudness_norm=True` is requested, unless they themselves have an invalid loudness reading.

**Environment overview**

- Environment location: Bare-metal (CPU)
- Method of NeMo install: from source (`pip install -e .`)

**Environment details**

- OS version: Ubuntu 24.04 (Linux 6.8, x86_64)
- PyTorch version: 2.13.0+cpu (via a CPU-only venv)
- Python version: 3.12.3

**Additional context**

`AudioAugmenter.augment_batch` (:419/:431) is called from `duplex_stt_dataset.py:307`, the live `DuplexSTTDataset` collation path, whenever the model config sets `use_room_ir_aug`/`use_mic_ir_aug` — both default `use_loudness_norm` to `True` via `cfg.get('roomir_use_loudness_norm', True)` / `micir_use_loudness_norm`. No shipped `examples/speechlm2/conf/*.yaml` currently sets `use_room_ir_aug`/`use_mic_ir_aug`, so this is a real, documented, opt-in training knob rather than a path the repo's own example configs exercise — but any training run that does enable room or mic IR augmentation loses loudness normalization for an entire batch the moment one row happens to be silent (a routine occurrence with padded/VAD-trimmed batches), silently degrading to RMS-only gain compensation.

`git log -S 'use_loudness_norm = False' -- nemo/collections/speechlm2/parts/augmentation.py` shows exactly one commit, `b9f2054e32` (#15092, "Implement Duplex Speech-to-text model") — present since the file's introduction, never touched or tested since.
